src/test/java/spoon/test/parent/ParentContractTest.java

```diff
@@ -117,6 +118,7 @@ public void testContract() throws Throwable {
 			if (o instanceof CtAnonymousExecutable && ("addParameter".equals(setter.getName()) || "setParameters".equals(setter.getName()))) continue;
 			if (o instanceof CtAnonymousExecutable && ("addThrownType".equals(setter.getName()) || "setThrownTypes".equals(setter.getName()))) continue;
 			if (o instanceof CtAnonymousExecutable && "setType".equals(setter.getName())) continue;
+			if (o instanceof CtTypeAccess && "setType".equals(setter.getName())) continue;
```

Comment:
OK.

It should also work if we set the parent, but it would be useless, right?

--Martin


On 01/05/2016 05:42 PM, Gérard Paligot wrote:
>
> In src/test/java/spoon/test/parent/ParentContractTest.java
> <https://github.com/INRIA/spoon/pull/470#discussion_r48865358>:
>
> > @@ -117,6 +118,7 @@ public void testContract() throws Throwable {
> >  			if (o instanceof CtAnonymousExecutable && ("addParameter".equals(setter.getName()) || "setParameters".equals(setter.getName()))) continue;
> >  			if (o instanceof CtAnonymousExecutable && ("addThrownType".equals(setter.getName()) || "setThrownTypes".equals(setter.getName()))) continue;
> >  			if (o instanceof CtAnonymousExecutable && "setType".equals(setter.getName())) continue;
> > +			if (o instanceof CtTypeAccess && "setType".equals(setter.getName())) continue;
>
> Yes, the setter is empty with |CtTypeAccess| now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/INRIA/spoon/pull/470/files#r48865358>.
>